examples/auth/src/getPathname.ts

```diff
@@ -0,0 +1,18 @@
+import { Location } from "react-router"
+
+const isObjectWithKey = <T extends string>(
+  given: unknown,
+  key: T
+): given is Partial<Record<T, unknown>> =>
+  typeof given === 'object' && given !== null && key in given
+
+export const getPathname = (location: Location): string | undefined => {
```

Comment:
Some variant of this function could also exist in the package itself for better community reuse but I wanted to POC it in the example first.